app/scripts/modules/core/src/slack/SlackReader.spec.ts

```diff
@@ -0,0 +1,38 @@
+import { API } from '@spinnaker/core';
+import { ISlackChannel, SlackReader } from './SlackReader';
+
+const mockSlackChannels: ISlackChannel[] = [
+  {
+    id: '0-test',
+    name: 'testchannel',
+    is_channel: true,
+    creator: 'creator',
+    is_archived: false,
+    name_normalized: 'testchannel',
+    num_members: 25,
+  },
+  {
+    id: '1-test',
+    name: 'fakechannel',
+    is_channel: true,
+    creator: 'creator2',
+    is_archived: false,
+    name_normalized: 'fakechannel',
+    num_members: 25,
+  },
+];
+
+describe('SlackReader', () => {
+  it('should fetch a list of public Slack channels', () => {
+    spyOn(SlackReader, 'getChannels').and.callThrough();
+    spyOn(API, 'one')
+      .and.callThrough()
+      .and.returnValue({ getList: () => Promise.resolve(mockSlackChannels) });
```

Comment:
@christopherthielen This was the best I could do using `callThrough` because I cannot spy on any of the private methods. I think it is valuable enough of a test to show that we have called `API` correctly, while still controlling the response. 

Another option is to utilize angular mocks like the other services, but I would prefer to reduce the dependency on angular as much as possible. 

Another

Reply:
Agreed that we should avoid using angular mocks in any new code.  this looks great to me, thanks for updating it!